combustion-android-ble/src/main/java/inc/combustion/framework/ble/BleDevice.kt

```diff
@@ -0,0 +1,253 @@
+/*
+ * Project: Combustion Inc. Android Framework
+ * File: Device.kt
+ * Author: https://github.com/miwright2
+ *
+ * MIT License
+ *
+ * Copyright (c) 2022. Combustion Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package inc.combustion.framework.ble
+
+import android.bluetooth.BluetoothAdapter
+import android.os.SystemClock
+import android.util.Log
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import com.juul.kable.*
+import inc.combustion.framework.LOG_TAG
+import inc.combustion.framework.service.DeviceConnectionState
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.onCompletion
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+
+/**
+ * Base class for Combustion Devices.
+ *
+ * @property mac Bluetooth MAC address of device.
+ * @property owner Owner of the instance's coroutine scope.
+ * @constructor Creates a new Device with the specified MAC.
+ *
+ * @param adapter Android BluetoothAdapter.
+ */
+internal open class BleDevice (
+    private val mac: String,
+    protected val owner: LifecycleOwner,
+    var advertisingData: LegacyProbeAdvertisingData,
+    adapter: BluetoothAdapter
+) {
+    companion object {
+        private const val PROBE_REMOTE_RSSI_POLL_RATE_MS = 2000L
```

Comment:
If these are the same across all devices, I'd suggest removing the `PROBE_` prefix.